docs/src/main/asciidoc/spring-cloud-commons.adoc

```diff
@@ -167,6 +167,8 @@ This feature addresses the problem of stateful beans that only get their configu
 For instance, if a `DataSource` has open connections when the database URL is changed via the `Environment`, you probably want the holders of those connections to be able to complete what they are doing.
 Then, the next time something borrows a connection from the pool, it gets one with the new URL.
 
+Sometimes, it might even be mandatory to apply `@RefreshScope` annotation on some beans which can be only initialized once. For example, the default Spring Boot DataSource is now `com.zaxxer.hikari.HikariDataSource`, which by default is `refreshable` (defined in `RefreshScopeBeanDefinitionEnhancer`), as HikariDataSource can no longer be modified once it's started. In order to refresh this kind of `immutable` beans, you will have to either annotate the beans with `@RefreshScope` or specify the list of class names under the property key `spring.cloud.refresh.extra-refreshable`.
```

Comment:
I would shorten the sentences a bit, so that each one conveys a single concept. Other than that, Ryan's modifications are on target. I'd fix a few lesser items, too. To do all of that, I'd use the following paragraph:

Sometimes, it might even be mandatory to apply `@RefreshScope` annotation on some beans which can be only initialized once. For example, the default Spring Boot DataSource is now `com.zaxxer.hikari.HikariDataSource`. By default, `HikariDataSource` is `refreshable` (you can set that in `RefreshScopeBeanDefinitionEnhancer`). However, HikariDataSource cannot be modified once it has been created. To refresh this kind of immutable bean, you have to either annotate the bean with @RefreshScope or specify the list of classname(s) under the property key (`spring.cloud.refresh.extra-refreshable`).

Reply:
@Buzzardo by "you can set that in RefreshScopeBeanDefinitionEnhancer", you mean "you can see that", right? Since it's a hard-coded in this class (I am not here to judge if it's a hack or not)..

Reply:
I thought you could set whether HikariDataSource is refreshable. Otherwise, why does it have a default? Where is it set?

Reply:
It MUST be refreshable, otherwise the app will get exceptions during the refreshing. It's hard-coded in RefreshScopeBeanDefinitionEnhancer. Please check out the issue #363 for which this PR is done.